### DISCUSSION TITLE
toggle event for DetailRowTemplate

### DISCUSSION THREAD

**Discussion by fastlink30** [dotnetcore/BootstrapBlazor#3194] [Q&A] [answered]
There are event to know when the DetailRowTemplate is opened?
also there are template to use when Items on the table are null/empty?

**Comment by ArgoZhang** [comment]
@fastlink30 In order to better understand your needs, more information is needed

**Reply by fastlink30** [reply]
```
<Table class="mt-2"
       TItem="SearchActionLogResponse"
       Items="@Items"
       IsBordered="true"
       IsStriped="true"
       AllowResizing="true" 
       IsFixedHeader=true
       IsAccordion=true
       HeaderStyle="TableHeaderStyle.Light"
       TableSize="TableSize.Compact"
       RenderMode="TableRenderMode.Auto">
    <TableColumns>
        <TableColumn @bind-Field="@context.ArteryId" Sortable="true" Filterable=true Width="140" />
        <TableColumn @bind-Field="@context.Controller" Sortable="true" Filterable=true Width="140" />
    </TableColumns>
    <DetailRowTemplate> <--- event when on UI is opened the child div of DetailRowTemplate?
        @{
            //ActionLogDetailResponse detailContext = utilityAPI.ActionLogDetail(context.Id);   
        }
        <div class="col-12"> <--- details inside here

        </div>
    </DetailRowTemplate>
</Table>
```

>  1 - i need to know when the **DetailRowTemplate** is opened on the UI, becouse i  need to get data details to use inside the **div**. blazor can't execute async (ActionLogDetail() is async) on @code, but if i have event on open **DetailRowTemplate** i can do this.

>  2 - if there are template (EmptyDataTemplate?) to use when Items associated to table are empty will be nice, this not force to use @if(items is null || !items.Any()) ....
ex:
```
<Table>
       TItem="SearchActionLogResponse"
       Items="@Items"
....
<TableColumns>
....
</TableColumns>
<EmptyDataTemplate>
 <Empty Image="https://gw.alipayobjects.com/zos/antfincdn/ZHrcdLPrvN/empty.svg" Width="133" Text="No data at this time" />
</EmptyDataTemplate>
</Table>
```

hope now is more clear :)

**Reply by ArgoZhang** [reply]
I have no idea what you're talking about. What do you need.

**Reply by ArgoZhang** [reply]
@fastlink30 sure! Let me see if it's possible to add a callback method for expanding and collapsing detail lines.

**Comment by ArgoZhang** [comment, marked as answer]
@fastlink30 for now there is a parameter 
```
/// <summary>
/// Get/Set Whether show empty data default false
/// </summary>
[Parameter]
public bool ShowEmpty { get; set; }
```

control whether to display an empty text or custom template use 

```
/// <summary>
/// Get/Set template for no data. default null
/// </summary>
[Parameter]
public RenderFragment? EmptyTemplate { get; set; }
```

same as your requirement `EmptyDataTemplate`

### The key question
```
[Parameter]
public Func<TItem, bool, Task>? OnToggleDetailRowCallback { get; set; }
```

first args `TItem` row context instance
second args `bool` indicate the row status is `expanded` or `collapsed`

link PR: https://github.com/dotnetcore/BootstrapBlazor/pull/3218?wt.mc_id=DT-MVP-5004174
